### DISCUSSION TITLE
Edit remote files via an SSH tunnel

### DISCUSSION THREAD

**Discussion by g24swint** [dbeaver/dbeaver#14480] [Ideas]
**Is your feature request related to a problem? Please describe.**
For most of my work, I actually like to keep my code repositories remote from my local machine. Therefore, editing remote files via a secure SSH connection is much preferred to storing them locally, and doing a git local commit/ local push/ remote pull to synchronize scripts

**Describe the solution you'd like**
I'd like the SQL editors to be able to open directories on a remote machine via SSH tunnels. This would be exactly analogous to the capability for remote editing that Visual Studio code offers.

**Describe alternatives you've considered**
An alternative would be scrapping my entrprise license of DBeaver and using a remote instance of PGAdmin for work

**Additional context**

**Comment by uslss** [comment]
thanks for suggestion

**Comment by p91paul** [comment]
Eclipse plugins should work in Dbeaver, so you could try this one https://stackoverflow.com/questions/95800/how-do-i-add-ftp-support-to-eclipse/6944100#6944100

**Comment by p91paul** [comment]
(beware, the link is outdated, kepler is a very old eclipse release, so you should search for an updated version)

**Comment by g24swint** [comment]
I tried the remote system explorer link but it only provides half the functionality. It seems like it only allows you to edit files, and something in the way the perspectives work keep you from being able to also execute the SQL. 

Since I'm an analyst, I'd really like to host all my code (python and SQL) remotely and run and save edited files there.  So, while this allows me to EDIT the remote file over sftp, it doesn't seem to then allow me to choose a connection and run an SQL command.

It's like when I'm the remote file editor it can't even "see" a database connection for the script to run against. In other words, even though in the DBeaver perspective I can open a connection, create a script, and run a file, I cannot in the remote-edit-file execute because it gives me an error of "no connection" and no connections appear in any of the usual places.

**Comment by p91paul** [comment]
On linux, sshfs would be an easy alternative, it allows you to mount a remote fs the same way you would mount an external usb disk. Possibly something similar exists for other OSes as well, and with this method not only Dbeaver but any application will be able to remotely edit files.
